### DISCUSSION TITLE
2022-16: 解决 Databend 命令行参数加载问题

### DISCUSSION THREAD

**Discussion by giscus** [Xuanwo/blog#349] [Announcements]
# 2022-16: 解决 Databend 命令行参数加载问题

An infrastructure engineer, focused on distributed storage system

https://xuanwo.io/reports/2022-16/

**Comment by pymongo** [comment]
config merge 配置文件+环境变量+命令行参数的问题我之前自己造的轮子是用「元编程」，通过过程宏/codegen 等方法自动生成三份代码(命令行参数、配置文件、环境变量) 但个人觉得 rust 元编程的支持有些不够用不足以完成像这样的逻辑，很多地方我过程宏都只能写死输出代码的内容
